### DISCUSSION TITLE
Is there a way to import an enum from rust in to slint?

### DISCUSSION THREAD

**Discussion by laycookie** [slint-ui/slint#6758] [Q&A] [answered]
Hello, I'm aware that if I define an enum in slint I'm able to use it in rust, but due to how my project is structured it would make more sense for me to define some structs in rust and then import them in to the `.slint` files when they are needed. Is there any way of doing that?

**Comment by ogoffart** [comment, marked as answer]
No, this is currently not possible.
The reason is that we want to keep the .slint file self-contained so that they can be previewed without having to compile the rust code.
We may support that in the future. This is tracked by issue https://github.com/slint-ui/slint/issues/1726
